aspnetcore/security/authentication/sociallogins.md

```diff
@@ -15,36 +15,48 @@ uid: security/authentication/sociallogins
+
+<img src="sociallogins/_static/facebook.svg" width="24">&nbsp;
+<img src="sociallogins/_static/twitter.svg" width="24">&nbsp;
+<img src="sociallogins/_static/google.svg" width="24">&nbsp;
+<img src="sociallogins/_static/microsoft.svg" width="24">
+
+Enabling users to login with their existing credentials is convenient for the users and shifts many of the complexities of managing the sign-in process onto a third party. For examples of how social logins can drive traffic and customer conversions, see case studies by [Facebook](https://developers.facebook.com/case-studies) and [Twitter](https://developers.facebook.com/case-studies/).
+
+> [!NOTE]
+> Packages presented here abstract a great deal of complexity of the OAuth 2.0 authentication flow, but understanding the details may become necessary when troubleshooting. Many resources are available; for example, see [Introduction to OAuth 2](https://www.digitalocean.com/community/tutorials/an-introduction-to-oauth-2) or [Understanding OAuth 2](http://www.bubblecode.net/en/2016/01/22/understanding-oauth2/). Some issues can be resolved by looking at [ASP.NET Core implementation](https://github.com/aspnet/Security/tree/dev/src).
 
 ## Create a New ASP.NET Core Project
 
 > [!NOTE]
 > The tutorial requires the latest version of Visual Studio 2015 and ASP.NET Core.
 
-* In Visual Studio, create a New Project (from the Start Page, or via **File > New > Project**)
+* In Visual Studio, create a New Project (from the Start Page, or via **File > New > Project**):
 
 ![image](sociallogins/_static/new-project.png)
 
-* Tap **Web Application** and verify **Authentication** is set to **Individual User Accounts**
+* Tap **Web Application** and verify **Authentication** is set to **Individual User Accounts**:
 
 ![image](sociallogins/_static/select-project.png)
 
-* Enable SSL
+## Enable SSL
 
-     * In solution explorer, right click the project and select **Properties**
+Some external authentication providers will reject requests coming from origins that don't use the **https** protocol. This reflects the trend of major providers like [Google](https://security.googleblog.com/2014/08/https-as-ranking-signal_6.html) moving their public API services to https and discontinuing the use of unencrypted endpoints. We encourage you to follow this trend and enable SSL for your entire site.
 
-     * On the left pane, tap **Debug**
+* In solution explorer, right click the project and select **Properties**.
 
-     * Check **Enable SSL**
+* On the left pane, tap **Debug**.
 
-     * Copy the SSL URL and paste it into the **App URL**
+* Check **Enable SSL**.
+
+* Copy the SSL URL and paste it into the **App URL**:
 
 ![image](sociallogins/_static/ssl.png)
 
-* Require SSL. Modify the services.AddMvc(); code in `Startup` under `ConfigureServices`:
+* Modify the services.AddMvc(); code in `Startup.cs` under `ConfigureServices` to reject all requests that are not coming over *https*:
 
 ````csharp
 services.AddMvc(options =>
```

Comment:
I hear that you will later have a dedicated page for Kestrel SSL, if not already. This whole section should be replaced with a link to that eventually. For now, adding MvcOptions.SslPort as a note.